Document Android compatibility. (#67)
okta/okta-jwt-verifier-java: README.md

- [Javadocs](https://developer.okta.com/okta-jwt-verifier-java/apidocs/)
- [Maven Central](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22com.okta.jwt%22%20a%3A%22okta-jwt-verifier%22)
- [Working With OAuth 2.0 Tokens](https://developer.okta.com/authentication-guide/tokens/)

# Android

> Okta JWT Verifier works with Android API 26+. If you have a use case for using this on Android and require compatibility with older Android versions please open an issue.
> 
> [Java 8 library desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring) may be required as Okta JWT Verifier makes use of java 8 features. See the link, or the below example on how to configure it.

Reviewer: While we do want to hear about these use cases, Java 7 is EOL, so it a little tricky.  We want to know about this so we could offer an android desugared lib if there is interest, but we also want to make sure folks can do this on their own until then.  Thoughts on something like:

```suggestion
Okta JWT Verifier makes use of Java 8 features.  These features work with Android API 26+. To make use of the Okta JWT Verifier with older versions of Android, you will need to ["desugar"]((https://developer.android.com/studio/write/java8-support#library-desugaring)) your application dependencies.
```

Author: There may be a miscommunication here. Desugaring works for some APIs, but not all. For instance `java.util.Base64` is not supported. So this sdk will not work at all in it's current form on older android APIs. 

Other java 8 APIs used in this sdk (like streams) are supported via desugaring though.

Reviewer: IIRC Bazel supports desugaring most of the Java 8 features, functional, time, etc. check with @FeiChen-okta he will know more

Reviewer 2: HI @bdemers

@JayNewstrom is correct. The latest desugaring library from google doesn't include Base64. Maybe in the future they will include it.

Reviewer: Ahh, I see, is the Base64 the only blocker? (If so we might be able to use the same Base64 impl that JJWT uses (which is a runtime dependency)

Author: @bdemers that was the only blocker. I did a `./mvnw install` and brought it into my android test project to try it and it worked appropriately on Android API 21. Thanks for the suggestion!

- [Javadocs](https://developer.okta.com/okta-jwt-verifier-java/apidocs/)
- [Maven Central](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22com.okta.jwt%22%20a%3A%22okta-jwt-verifier%22)
- [Working With OAuth 2.0 Tokens](https://developer.okta.com/authentication-guide/tokens/)

# Android

Okta JWT Verifier works with Android API 26+. If you have a use case for using this on Android and require 

> compatibility with older Android versions please open an issue. 

Reviewer: We may want to point to desugaring documentation for older Android versions. @FeiChen-okta might be able to point you in the right direction

Author: Sure, I can add that here as well.
